### ISSUE TITLE
Implement mechanical vignetting

### ISSUE THREAD

**Issue by JorisDeRidder** [IvS-KULeuven/PlatoSim3#334]
Currently only geometric vignetting is implemented. It would be nice to also have mechanical vignetting. This requires:

1. Refactoring the current vignetting code to make the distinction between geometric and mechanical vignetting possible
2. Adapt the inputfile.yaml file.
3. Add mechanical vignetting to the function Detector::generateThroughputMap()

**Comment by SaraRegibo** [comment]
Implement as

```
if(angular distance for pixel (row, column) from OA >- 18.8876)
     throughputMap(row, column) = 0
```

So no flux is accumulated in these pixels (beyond the edge of the FOV) during exposure.  There is, however, flux accumulation during readout (background), leakage of flux to these pixels due to blooming, etc.  Which effects contribute is still TBD.

**Comment by SaraRegibo** [comment]
Proposed structure for the configuration file (in the `CCD` block):
```
    Vignetting:
        NaturalVignetting: 
            ExpectedValue:           0.904           # Expected value of the throughput efficiency due to vignetting
        MechanicalVignetting:
            RadiusFOV:               18.8876         # Radius of the FOV [degrees]

    IncludeNaturalVignetting:        yes             # Include brightness attenuation towards the edge of the FOV due to natural vignetting
    IncludeMechanicalVignetting:     yes             # Include blockage of incoming flux during exposure at the edge of the FOV due to mechanical vignetting
```

What needs to be adapted?
* implementation in `Detector` (+ revisit the implementation of all other effects to check whether they should be updated; e.g. accumulation of background flux during readout).
* all input files
* Python scripts
* tests
* tutorials
* TNs
* documentation

**Comment by SaraRegibo** [comment]
Closed by Pull Request #341
